graphdatascience/query_runner/cypher_graph_constructor.py

```diff
@@ -278,46 +306,52 @@ def collect_to_dict(row: Dict[str, Any]) -> Dict[str, Any]:
                     return {column: row[column] for column in schema.rels_per_df[i].properties}
 
                 rel_dict_df = DataFrame(rel_dict)
-                rel_dict_df[CypherAggregationApi.REL_PROPERTIES] = df.apply(collect_to_dict, axis=1)
-                rel_dict_df[CypherAggregationApi.REL_PROPERTIES + self._BIT_COL_SUFFIX] = True
+                rel_dict_df[properties_key] = df.apply(collect_to_dict, axis=1)
+                rel_dict_df[properties_key + self._BIT_COL_SUFFIX] = True
                 rel_dict_df[CypherAggregationApi.SOURCE_NODE_PROPERTIES] = None
                 rel_dict_df[CypherAggregationApi.SOURCE_NODE_PROPERTIES + self._BIT_COL_SUFFIX] = False
 
                 adjusted_dfs.append(rel_dict_df)
 
             return adjusted_dfs
 
-        def nodes_config(self, node_cols: List[EntityColumnSchema]) -> str:
+        def nodes_config_part(self, node_cols: List[EntityColumnSchema], is_cypher_projection_v2: bool) -> List[str]:
             # Cannot use a dictionary as we need to refer to the `data` variable in the cypher query.
             # Otherwise we would just pass a string such as `data[0]`
             nodes_config_fields: List[str] = []
-            if reduce(lambda x, y: x | y.has_labels(), node_cols, False):
+            if any(x.has_labels() for x in node_cols):
                 nodes_config_fields.append(
                     f"{CypherAggregationApi.SOURCE_NODE_LABEL}: {CypherAggregationApi.SOURCE_NODE_LABEL}"
                 )
+                if is_cypher_projection_v2:
+                    nodes_config_fields.append(
+                        f"{CypherAggregationApi.TARGET_NODE_LABEL}: NULL",
+                    )
```

Comment:
Any reason that we didn't set targetNodeLabels before? Should we? Or are we sure that all nodes with labels appear at least once in the sourceNode position?

Reply:
cool, thanks!